webrender/src/display_list_flattener.rs

```diff
@@ -1552,13 +1559,13 @@ impl<'a> DisplayListFlattener<'a> {
             self.shadow_prim_stack[idx].1.push((prim_index, clip_and_scroll));
         }
 
-        let prim_index = self.create_primitive(
```

Comment:
Hrm. Out of curiosity, does this change allow for making this optimization or is it an unrelated change?

Reply:
This change allows it - previously we would reference the same primitive in the slow shadow path below. Now we build a separate primitive there. There's a slight memory overhead to this for slow text shadows, which we might be able to improve on in the future, but it should be minor.